From 21c95afdc22b2c437468fc203f1f7c5abc08a6f9 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Wed, 11 Jul 2012 16:39:08 +0800
Subject: [PATCH 308/635] ENGR00216843 testPreviewFpsRange failed in DL with
 25% possibility

The case need strict frame interval time according the fps setting
The failure case is caused by some time there is not buffer in ready_q
queue in mxc_v4l_capture driver.
The buffer location is as below:
two buffers in Surfaceflinger, one for composition, the other is in ready queue
one buffer in driver's working_q for csi interrupt handling
one buffer in driver's done_q, which can be dequeue by Camera HAL
one buffer in Camera HAL, which just dequeued from v4l driver, and not queue
to SurfaceFlinger yet
mxc v4l capture driver need ready_q have at least one buffer at any time of
irq handler
if not, there next irq will only update the dummy buffer

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 mx6/libcamera/CameraHal.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/mx6/libcamera/CameraHal.h b/mx6/libcamera/CameraHal.h
index bb24432..10a5bc5 100755
--- a/mx6/libcamera/CameraHal.h
+++ b/mx6/libcamera/CameraHal.h
@@ -55,12 +55,12 @@
 #define V4LSTREAM_WAKE_LOCK "V4LCapture"
 #define MAX_SENSOR_NAME 32
 
-#define PREVIEW_HEAP_BUF_NUM    5
-#define VIDEO_OUTPUT_BUFFER_NUM 5
-#define POST_PROCESS_BUFFER_NUM 5
-#define TAKE_PIC_QUE_BUF_NUM 5
+#define PREVIEW_HEAP_BUF_NUM    6
+#define VIDEO_OUTPUT_BUFFER_NUM 6
+#define POST_PROCESS_BUFFER_NUM 6
+#define TAKE_PIC_QUE_BUF_NUM 6
 
-#define PREVIEW_CAPTURE_BUFFER_NUM 5
+#define PREVIEW_CAPTURE_BUFFER_NUM 6
 #define PICTURE_CAPTURE_BUFFER_NUM 3
 
 #define DEFAULT_PREVIEW_FPS (15)
-- 
1.8.0

